### PR TITLE
Saving form information when user go to another modal

### PR DESCRIPTION
**Pull request by YaroslavOvdii**
@tonytlwu @squallstar 
## Issue
https://github.com/Fliplet/fliplet-studio/issues/3755

## Description
Created a save function witch is saving data on change icon or splash screen.

## Screenshots/screencasts
issue-3755-google

## Backward compatibility

This change is fully backward compatible.

## Notes
Slightly altered saving functions so that they return a promise.

### CODE REVIEW

File: js/interface.js [hunk 1]
```diff
@@ -686,36 +698,41 @@ $('input[type="file"]').on('change', function() {
...
+    Fliplet.Studio.emit('overlay', {
+      name: 'app-settings',
+      options: {
+        size: 'large',
+        title: 'App Settings',
+        section: 'appSettingsGeneral',
+        appId: Fliplet.Env.get('appId')
+      }
+    });
   });
```
Comment (tonytlwu) [inline js/interface.js]: There should be a `.catch()` clause like what you've done for iOS AAB UI.

Comment (YaroslavOvdii) [reply js/interface.js]: @tonytlwu 
Fixed

File: js/interface.js [hunk 2]
```diff
@@ -686,36 +698,45 @@ $('input[type="file"]').on('change', function() {
...
+        size: 'large',
+        title: 'App Settings',
+        section: 'appSettingsGeneral',
+        appId: Fliplet.Env.get('appId')
+      }
+    });
+  }).catch(function (error) {
+    Fliplet.Modal.alert({
+      message: Fliplet.parseError(error)
+    })
```
Comment (tonytlwu) [inline js/interface.js]:
```suggestion
    });
```

File: js/interface.js [hunk 3]
```diff
@@ -686,36 +698,45 @@ $('input[type="file"]').on('change', function() {
...
+    Fliplet.Studio.emit('overlay', {
+      name: 'app-settings',
+      options: {
+        size: 'large',
+        title: 'App Settings',
+        section: 'launchAssets',
+        appId: Fliplet.Env.get('appId')
+      }
+    });
   });
```
Comment (tonytlwu) [inline js/interface.js]: There should be a `.catch()` clause here too.

File: js/interface.js (hunk 2 above)
Comment (YaroslavOvdii) [reply js/interface.js]: @tonytlwu 
Added semicolons.

File: js/interface.js (hunk 3 above)
Comment (YaroslavOvdii) [reply js/interface.js]: @tonytlwu 
Sorry I didn't notice that.
Fixed.

**Pull request merged**